src/producer/mod.rs

```diff
@@ -93,38 +96,37 @@ impl Producer<TcpStream> {
 
 impl<S: Read + Write> Producer<S> {
     /// Connect to a Faktory server with a non-standard stream.
-    pub fn connect_with(stream: S, pwd: Option<String>) -> Result<Producer<S>, Error> {
-        Ok(Producer {
-            c: Client::new_producer(stream, pwd)?,
-        })
+    pub fn connect_with(stream: S, pwd: Option<String>) -> Result<Producer<S>> {
+        let c = Client::new_producer(stream, pwd)?;
+        Ok(Producer { c })
```

Comment:
Why change the body structure here?

Reply:
Oh, I think that was left over from something I was doing earlier, changing it back.